client/src/common/notebook.ts

```diff
@@ -427,6 +441,10 @@ class NotebookDocumentSyncFeatureProvider implements NotebookDocumentSyncFeature
 	private readonly notebookDidOpen: Set<string>;
 	private readonly disposables: vscode.Disposable[];
 	private readonly selector: vscode.DocumentSelector;
+	private readonly _onChangeNotificationSent: vscode.EventEmitter<VNotebookDocumentChangeEvent>;
```

Comment:
Do we really need event emitters here. As I understand it and compared to the text document synchronization client will hook the listeners on the NotebookDocumentSyncFeature. Since a NotebookDocumentSyncFeatureProvider belongs to exactly one NotebookDocumentSyncFeature can we not simply use a callback functions that the NotebookDocumentSyncFeature passes into the NotebookDocumentSyncFeatureProvider?

Reply:
Callbacks work too. I'll change it.